examples/redismq-java-client/test/redismq_java_client/example_test.clj

```diff
@@ -1,32 +1,84 @@
+            [clojure.spec :as cs]
+            [clojure.test.check.clojure-test :refer [defspec]]
+            [clojure.test.check.generators :as gen]
+            [clojure.test.check.properties :refer [for-all]]
+            [ctim.events.schemas :refer [CreateEventType]]
+            [ctim.generators.id :as gen-id]
+            [ctim.schemas
+             [actor :refer [StoredActor]]
+             [coa :refer [StoredCOA] :as coa]]
+            [flanders.spec :as fs]
             [redismq.core :as rmq]
-            [schema.core :as s])
+            [schema.core :as s]
+            [taoensso.carmine :as car])
   (:import [client Example]))
 
-(deftest test-example
-  (let [{{:keys [host port queue-name timeout-ms]} :redis}
-        (->> (mp/read-property-files ["default-example.properties"
-                                      "example.properties"])
-             mp/transform
-             (mp/coerce-properties {:redis {:host s/Str
-                                            :port s/Int
-                                            :timeout-ms s/Int
-                                            :queue-name s/Str}}))
-
-        queue
-        (rmq/make-queue queue-name
-                        {:host host
-                         :port port
-                         :timeout-ms timeout-ms}
-                        {:max-depth 0})]
-
-    (rmq/enqueue queue {:foo "bar"})
-
-    (testing "Can read from the message queue"
-      (is (= {:foo "bar"}
-             (some-> (Example/getAnEvent host port queue-name)
-                     (json/parse-string keyword)
-                     :data))))))
+(cs/def ::actor (fs/->spec StoredActor "stored-actor"))
+(cs/def ::coa (fs/->spec StoredCOA "stored-coa"))
+
+(defn read-config []
+  (->> (mp/read-property-files ["default-example.properties"
+                                "example.properties"])
+       mp/transform
+       (mp/coerce-properties {:redis {:host s/Str
+                                      :port s/Int
+                                      :timeout-ms s/Int
+                                      :queue-name s/Str}})))
+
+(defn build-queue [{{:keys [host port queue-name timeout-ms]} :redis
+                    :as _config_}]
+  (rmq/make-queue queue-name
+                  {:host host
+                   :port port
+                   :timeout-ms timeout-ms}
+                  {:max-depth 100}))
+
+(def gen-actor-create-event
+  (gen/fmap (fn [[actor event-id]]
+              {:entity actor
+               :type CreateEventType
+               :id event-id})
+            (gen/tuple (cs/gen ::actor)
+                       (gen-id/gen-short-id-of-type :actor))))
+
+(def gen-coa-create-event
+  (gen/fmap (fn [[coa event-id]]
+              {:entity coa
+               :type CreateEventType
+               :id event-id})
+            (gen/tuple (cs/gen ::coa)
+                       (gen-id/gen-short-id-of-type :event))))
+
+(def gen-some-events
```

Comment:
Actually this is a generator that returns a vector, so I think a verb is appropriate.